bears/general/CasingBear.py

```diff
@@ -0,0 +1,134 @@
+import re
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.bearlib.naming_conventions import (
+    to_camelcase, to_pascalcase, to_snakecase)
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+
+
+class CasingBear(LocalBear):
+
+    def run(self,
+            filename,
+            file,
+            casing: str,
+            language: str):
+        """
+        Checks whether all identifier names (variables, classes, objects)
+        follow a certain naming convention.
+
+        :param casing:   camelCasing or snake_casing or PascalCasing
+        :param language: The language of the file, which is used to determine
+                         the keywords to ignore.
+        """
+        casing_convention = {"camel": to_camelcase,
+                             "pascal": to_pascalcase,
+                             "snake": to_snakecase}
+
+        if casing not in casing_convention:
+            self.err("Invalid file-naming-convention provided: " + casing)
+            return
+
+        self.convertor = casing_convention[casing]
+        coalang = LanguageDefinition(language=language)
+
+        if "keywords" not in coalang or "special_chars" not in coalang:
+            self.warn("CasingBear did not run because coalang definition "
+                      "is not available for " + language)
+            return
+
+        self.keywords = coalang["keywords"]
+        self.delim = re.escape(str(coalang["special_chars"]) + " \t\n")
+
+        string_enclosers = {"\"": "\"",
```

Comment:
also this function does not account for strings that would have started withing comments like this would also return this result 
```
# A string within a "comment"
```
this function would return also return "comment" but that's not a valid string right? since it starts within a comment